### PR TITLE
[refactor] Remove boost mutex, use std::mutex instead

### PR DESCRIPTION
**Pull request by yangzhg**
## Proposed changes

Remove boost mutex, use std::mutex instead

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [x] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (#5575) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/common/object_pool.h [hunk 1]
```diff
@@ -18,8 +18,8 @@
 #ifndef DORIS_BE_SRC_COMMON_COMMON_OBJECT_POOL_H
 #define DORIS_BE_SRC_COMMON_COMMON_OBJECT_POOL_H
 
-#include <boost/thread/locks.hpp>
-#include <boost/thread/mutex.hpp>
+
```
Comment (stdpain) [inline be/src/common/object_pool.h:21]: code format here ?

File: be/src/runtime/data_stream_recvr.cc [hunk 2]
```diff
@@ -19,8 +19,8 @@
 
 #include <google/protobuf/stubs/common.h>
 
-#include <boost/thread/locks.hpp>
-#include <boost/thread/mutex.hpp>
+
```
Comment (stdpain) [inline be/src/runtime/data_stream_recvr.cc:22]: code format here

File: be/src/runtime/thread_resource_mgr.h [hunk 3]
```diff
@@ -23,7 +23,7 @@
 #include <boost/function.hpp>
 #include <boost/scoped_ptr.hpp>
 #include <boost/shared_ptr.hpp>
-#include <boost/thread/mutex.hpp>
+#include <mutex>
```
Comment (stdpain) [inline be/src/runtime/thread_resource_mgr.h:26]: adjust include file order ?

File: be/src/runtime/cache/cache_utils.h [hunk 4]
```diff
@@ -35,7 +35,7 @@
 namespace doris {
 
 typedef boost::shared_lock<boost::shared_mutex> CacheReadLock;
```
Comment (morningman) [inline be/src/runtime/cache/cache_utils.h:37]: shared_lock can be replaced too?

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**